molgenis-data-mysql/src/main/java/org/molgenis/data/mysql/MysqlRepository.java

```diff
@@ -956,32 +968,7 @@ protected String getSortSql(Query q, List<Object> parameters)
 
 			if (sortSql.length() > 0) sortSql = new StringBuilder("ORDER BY ").append(sortSql.substring(2));
 		}
-		// in case of a findAll(Iterable<Object> ids) MySQL returns matching rows in a random order. use ORDER BY FIELD
-		// to guarantee the right order. This is required by the Repository interface.
-		else if (q.getRules() != null && q.getRules().size() == 1)
-		{
-			QueryRule queryRule = q.getRules().iterator().next();
-			if (queryRule.getOperator() == Operator.IN && queryRule.getValue() instanceof Iterable<?>)
-			{
-				Iterable<?> values = (Iterable<?>) queryRule.getValue();
-				if (values != null && values.iterator().hasNext())
-				{
-					// use prepared statements for the values
-					String fieldIds = StreamSupport.stream(values.spliterator(), false).map(value -> "?")
-							.collect(Collectors.joining(","));
-
-					// define order of returned rows
-					sortSql = new StringBuilder("ORDER BY FIELD(").append("this.").append('`')
-							.append(queryRule.getField()).append('`').append(",").append(fieldIds).append(")");
-
-					// use prepared statements for the values
-					parameters.addAll(StreamSupport.stream(values.spliterator(), false).map(value -> value.toString())
-							.collect(Collectors.toList()));
-				}
-			}
-		}
```

Comment:
This is the old solution, it specifies the order of the result rows in an ORDER BY FIELD clause. In MySQL 5.1 it runs into a bug where the collected MREF IDs get truncated if you add an ORDER BY.